src/c-writer.cc

```diff
@@ -31,6 +31,9 @@
 #include "wabt/stream.h"
 #include "wabt/string-util.h"
 
+#define WASM_SPECIAL_PREFIX "wasm_"
+#define INTERNAL_MAIN_NAME "__main_argc_argv"
```

Comment:
I am guessing this can become just "main" if you drop arguments... which is annoying.

Reply:
I think we should probably create some kind `crt1.c` that is responsible for calling main... that would then define the entry point (normally called `_start`). 

Then we would include a `main.c` in the final native build (like we do for `printf.c`) which calls the `_start` function.

The `_start` function could take argc and argv unconditionally, and it would be in charge of calling `__wasm_call_ctors` too.

Reply:
Yeah, I'm not quite sure where that name __main_argc_argv actually comes from (i.e., why it isn't just "main" in the first place).

The C compiler moans a lot if "main" does not have the right types, though.  So this two-level thing lets us include the necessary casts.

Reply:
I agree that eventually we need to make all this more robust.  Generating a "main" on the fly should get us over the hump and simplify everybody's life when they try to replicate my steps.

Reply:
main gets name mangles to `__main_argc_argv` by llvm (because wasm doesn't support more than one type for a given symbol and main has at least two possible signatures.

The other version of main with no args is defined with the `__main_void` alias.

See https://github.com/WebAssembly/wasi-libc/blob/main/libc-bottom-half/crt/crt1-command.c and https://github.com/WebAssembly/wasi-libc/blob/main/libc-bottom-half/sources/__main_void.c for how to deal with both possible signatures.